main.tf

```
@@ -1,13 +1,103 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled = module.this.enabled
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  records = try(jsondecode(var.records), var.records)
+  record_objects = {
+    for i, r in local.records :
+    join("_", compact([
+      try(lower(r.name), ""),
+      lower(r.type),
+      try(lower(r.set_identifier), ""),
+      try(lower(r.failover), "")
+      ])) => {
+      name = try(lower(r.name), "")
+      type = r.type
+      ttl  = try(r.ttl, null)
+      records = try(r.records, null)
+      alias = {
+        name                   = try(r.alias.name, null)
+        zone_id                = try(r.alias.zone_id, null)
+        evaluate_target_health = try(r.alias.evaluate_target_health, null)
+      }
+      geolocation = {
+        continent   = try(r.geolocation.continent, null)
+        country     = try(r.geolocation.country, null)
+        subdivision = try(r.geolocation.subdivision, null)
+      }
+      allow_overwrite = try(r.allow_overwrite, var.allow_overwrite)
+      health_check_id = try(r.health_check_id, null)
+      set_identifier  = try(r.set_identifier, null)
+      weight          = try(r.weight, null)
+      failover        = try(r.failover, null)
+    }
   }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+resource "aws_route53_zone" "default" {
+  count = local.enabled ? 1 : 0
+  
+  name          = var.zone_name
+  comment       = module.this.name
+  force_destroy = var.force_destroy
+
+  tags = module.this.tags
+}
+
+resource "aws_route53_record" "default" {
+  for_each = local.record_objects
+
+  zone_id         = aws_route53_zone.default[0].zone_id
+  type            = each.value.type
+  name            = each.value.name
+  allow_overwrite = var.allow_overwrite
+  health_check_id = try(each.value.health_check_id, null)
+  set_identifier  = try(each.value.set_identifier, null)
+
+  # If TTL didn't set and the record type is not an alias - use default TTL
+  ttl = each.value.alias.name == null ? coalesce(each.value.ttl, var.default_ttl) : null
+
+  # Split TXT records which have a size of more than 255 chars
+  records = each.value.records != null ? [for r in each.value.records :
+    each.value.type == "TXT" && length(regexall("(\\\"\\\")", r)) == 0 ?
```

I don't think this regex expresses what it is intended to express.

 What are we trying to do with this regex? `"(\\\"\\\")"`

```suggestion
    each.value.type == "TXT" && length(regexall("(\\\"\\\")", r)) == 0 ?
```

---

It is just checking if a record isn't already split.

---

Aha, ok

---

This could be added as a comment for future contributors

---

@osterman This regex etc. is OK.